File: Python/1196.py
import bisect
import sys

nrecords = int(sys.stdin.readline())
teacher_years = [int(sys.stdin.readline()) for _ in range(nrecords)]

years = int(sys.stdin.readline())
count = 0
for _ in range(years):
    year = int(sys.stdin.readline())
    index = bisect.bisect_left(teacher_years, year)
    if index != len(teacher_years) and teacher_years[index] == year:
        count += 1
print(count)

File: C++/1196.cpp
#include <algorithm>
#include <iostream>
#include <iterator>
#include <vector>

int main() {
  std::ios_base::sync_with_stdio(false);
  std::cin.tie(nullptr);

  int nrecords;
  std::cin >> nrecords;
  std::vector<int> teacher_years;
  teacher_years.reserve(nrecords);
  std::copy_n(std::istream_iterator<int>(std::cin), nrecords,
              std::back_inserter<std::vector<int>>(teacher_years));

  int years;
  std::cin >> years;
  int count = 0;
  for (int i = 0; i < years; ++i) {
    int year;
    std::cin >> year;
    if (std::binary_search(teacher_years.begin(), teacher_years.end(), year)) {
      ++count;
    }
  }
  std::cout << count << '\n';

  return 0;
}
